src/components/util/remote-ressource-handler.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+export class RemoteRessourceHandler {
+    fetcher = undefined;
+    errorHandler = undefined;
+    postUpdate = undefined;
+
+    constructor(fetcher, errorHandler, postUpdate) {
+        this.fetcher = fetcher;
+        this.errorHandler = errorHandler;
+        this.postUpdate = postUpdate;
+    }
+
+    values = undefined;
+    updating = undefined;
+
+    cbUpdateDone = new Set();
+
+    get(cbUpdateDone) {
+        if (this.values === undefined) {
+            if (cbUpdateDone) this.cbUpdateDone.add(cbUpdateDone);
+            if (!this.updating) {
+                this.updating = true;
+                Promise.all([this.fetcher()])
+                    .then((val) => {
+                        this.values = val[0];
+                        if (this.postUpdate) this.postUpdate(this.values);
+                        this.updating = false;
+                        this.cbUpdateDone.forEach((cb) => cb());
+                        this.cbUpdateDone.clear();
+                    })
+                    .catch((error) => {
+                        if (this.errorHandler) this.errorHandler(error);
+                        this.values = {};
+                    });
+            }
+            return undefined;
+        }
+        if (cbUpdateDone) cbUpdateDone();
+        return this.values;
+    }
+
+    length(cbUpdateDone) {
```

Comment:
rename to lengthOrFetch? lengthOtherwiseFetch ?

Reply:
done